src/hotspot/share/gc/shared/gc_globals.hpp

```diff
@@ -412,7 +412,7 @@
                                                                             \
   product(uintx, InitialSurvivorRatio, 8,                                   \
           "Initial ratio of young generation/survivor space size")          \
-          range(0, max_uintx)                                               \
```

Comment:
There is code somewhere which sets InitialSurvivorRatio to 3 if it is smaller than that. It should be removed. Somewhere around `parallelArguments.cpp:108).
There is similar code next to it for `MinSurvivorRatio` which is dead code too (`MinSurvivorRatio` is already bounded with 3 at minimum).
Also, previously this value has been overridden silently, bailing out is a behavioral change that requires a CSR.